examples/tests/valid/intrinsics.test.w

```diff
@@ -0,0 +1,23 @@
+bring fs;
+bring expect;
+bring "./subdir/bar.w" as bar;
+
+let path = "SHOULD_IGNORE";
+let filename = "intrinsics.test.w";
+let currentFile = @path(filename);
+expect.equal(fs.basename(currentFile), filename);
+
+let currentDir = @path(".");
```

Comment:
TBH I am not sure why is this better than `__dirname`. I am assuming we are not checking that the referenced path actually exists, so effectively this is just `join(__dirname, ...)`. 

What am I missing?

I can see the benefit of something like `@asset()` which can be lifted and all but if this is just to reference something relative to the current directory of the source file, I don't think that's the right design.

I would also say the name is somewhat misleading and doesn't imply that this is about the current source file. 

My recommendation is to just go with `@dirname` and keep it simple and low surface for now and introduce something like `@asset()` if we really want to do the right thing.

Reply:
My issues with `@dirname`:
1. It would have to be `@dirname()` based the reasoning I proved in the other rfc
2. 99% percent of the time, you would immediately `join(@dirname(), "something")`, so let's design for the common case
3. This API mimics other types of "relative path" APIs we're going to have. `@asset()` `@inflight()` will look similar.
4. `@asset()` has more implications than `@path()` imo so I think it can coexist
5. This forces paths to be resolved upfront which should help to remind users that the path they are referencing is special.

> I would also say the name is somewhat misleading and doesn't imply that this is about the current source file.

This is fair, I'm open to alternative name but I still think the API is valuable. Any other names I could think of were either too long or ambiguous. `@resolve()` is the only other idea I kind of like but it implies there is more interesting resolution logic like node's. `@asAbsolute()` is another idea.

Also, I think the main reason that `@dirname()` doesn't seem misleading is because of our familiarity with JS. I can easily see that being ambiguously referring to the cwd.

> keep it simple and low surface

You've mentioned simplicity a few times and I don't see what is complicated about `@path()`. As you said it's basically just dirname + join and the implementation is already complete. The surface area is something that will have to be deal with with other intrinsic so it will soon be well trodden.

Reply:
> 99% percent of the time, you would immediately join(@dirname(), "something"), so let's design for the common case


I did a search in GitHub (public repositories) of all the occurrence of __dirname and noticed that in all occasions __dirname was used with join , so I think that @path() makes a lot of sense. Just as a side note , I have conducted this search a couple of days ago to prove to Mark that we need __dirname and @path but have seen that I am probably wrong. @path(.) will probably not be used a lot

Reply:
I also , like the @inflight(), I feel it sets us up for better cross language support

Reply:
I can see the argument for either side, `@path("./foo")` seems like an ok shorthand for `fs.join(@dirname, "./foo")` but I'm also ok with leaving it out for now if there are ambiguity issues. @MarkMcCulloh It's your call

Reply:
reworked to `@dirname`